### PR TITLE
Removes_valid_input_amounts

### PR DESCRIPTION
**Pull request by cesve**
### Changed
Removed valid input amounts.

### Fixed

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by RasmusBurge-CG** [comment]
Removed all test that checks VALID_AMOUNTS_NEEDED

**Comment by RasmusBurge-CG** [comment]
Screenshot 2022-05-17 at 10 25 37

**Comment by RasmusBurge-CG** [comment]
Could you test this? @cesve

**Comment by cesve** [comment]
Tested on stage 22-05-17:

Script worked when different values of "Amount needed (ng)":

Screenshot 2022-05-17 at 19 18 26

The calculations were correct:
Screenshot 2022-05-17 at 19 18 34

**Comment by cesve** [comment]
Tested on stage 22-05-17:

The script worked well when different values were set for UDF "Amount needed (ng)".
Screenshot 2022-05-17 at 19 18 34

**Comment by cesve** [comment]
@RasmusBurge-CG it looks really good!

**Comment by RasmusBurge-CG** [comment]
@cesve are we ready for review? :)

**Comment by cesve** [comment]
Yes, @Karl-Svard when you have time it would be great if you can review :)

File: cg_lims/EPPs/udf/calculate/calculate_water_volume_rna.py [hunk 1]
```diff
@@ -29,11 +28,6 @@ def calculate_sample_and_water_volumes(artifacts: List[Artifact]):
             )
             missing_udfs += 1
             continue
-        if amount_needed not in VALID_AMOUNTS_NEEDED:
-            raise InvalidValueError(
-                f"'Amount needed (ng)' missing or incorrect for one or more samples. Value can "
-                f"only be {', '.join(map(str, VALID_AMOUNTS_NEEDED))}."
-            )
```
Comment (Karl-Svard) [inline cg_lims/EPPs/udf/calculate/calculate_water_volume_rna.py:36]: Keep this check if we still require the amount_needed udf to exist. Just modify it to work similar to the one that checks the concentration.

File: tests/EPPs/udf/calculate/test_calculate_water_volume_rna.py [hunk 2]
```diff
@@ -22,57 +22,6 @@ def test_calculate_water_volume_rna_missing_concentration(artifact_1: Artifact):
...
-
-    # WHEN calculating sample and water volumes for all samples
-    with pytest.raises(InvalidValueError) as error_message:
-        calculate_sample_and_water_volumes(artifacts)
-
-    # THEN the InvalidValueError exception should be raised
-    assert (
-        f"'Amount needed (ng)' missing or incorrect for one or more samples."
-        in error_message.value.message
-    )
```
Comment (Karl-Svard) [inline tests/EPPs/udf/calculate/test_calculate_water_volume_rna.py:48]: Keep this pytest as it will still be relevant.

Comment (Karl-Svard) [inline tests/EPPs/udf/calculate/test_calculate_water_volume_rna.py:48]: Keep this pytest as it will still be relevant.

**Review by Karl-Svard — CHANGES_REQUESTED** [review]

File: cg_lims/EPPs/udf/calculate/calculate_water_volume_rna.py (hunk 1 above)
Comment (cesve) [reply cg_lims/EPPs/udf/calculate/calculate_water_volume_rna.py:36]: ok

File: tests/EPPs/udf/calculate/test_calculate_water_volume_rna.py [hunk 3]
```diff
@@ -22,57 +22,6 @@ def test_calculate_water_volume_rna_missing_concentration(artifact_1: Artifact):
         in error_message.value.message
     )
```
Comment (cesve) [inline tests/EPPs/udf/calculate/test_calculate_water_volume_rna.py:24]:
```suggestion

@pytest.mark.parametrize(
    "concentration",
    [
        100,
    ],
)
def test_calculate_water_volume_rna_amount_needed_missing(
    concentration, artifact_1: Artifact
):
    # GIVEN a list of artifacts with one artifact having no 'Amount Needed (ng)' udf
    artifact_1.udf["Concentration"] = concentration
    del artifact_1.udf["Amount needed (ng)"]
    artifacts = [artifact_1]

    # WHEN calculating sample and water volumes for all samples
    with pytest.raises(InvalidValueError) as error_message:
        calculate_sample_and_water_volumes(artifacts)

    # THEN the InvalidValueError exception should be raised
    assert (
        f"'Amount needed (ng)' missing or incorrect for one or more samples."
        in error_message.value.message
```

**Comment by RasmusBurge-CG** [comment]
@Karl-Svard redo för review när du har tid :)

File: tests/EPPs/udf/calculate/test_calculate_water_volume_rna.py [hunk 4]
```diff
@@ -33,45 +30,18 @@ def test_calculate_water_volume_rna_amount_needed_missing(
...
         calculate_sample_and_water_volumes(artifacts)
 
     # THEN the InvalidValueError exception should be raised
     assert (
-        f"'Amount needed (ng)' missing or incorrect for one or more samples."
-        in error_message.value.message
-    )
-
+        f"Could not apply calculation for 1 out of 1 sample(s): 'Concentration' or 'Amount needed (ng)' is missing!"
+        in error_message.value.message)
```
Comment (Karl-Svard) [inline tests/EPPs/udf/calculate/test_calculate_water_volume_rna.py:44]:
```suggestion
        in error_message.value.message
    )
```

**Review by Karl-Svard — APPROVED** [review]
Nice! Litet förslag som gör koden snyggare bara :)

**Comment by RasmusBurge-CG** [comment]
Screenshot 2022-05-31 at 10 52 52

Test error messages for when 'Amount needed (ng)' is none.

**Comment by cesve** [comment]
Should I re-do the test I did 2 weeks ago?

**Comment by RasmusBurge-CG** [comment]
> Should I re-do the test I did 2 weeks ago? @RasmusBurge-CG

Yea, lets do that. It should not be any difference but you never know. :)

**Comment by cesve** [comment]
@RasmusBurge-CG test is done successfully! The updated script is ready for production.

**Pull request merged**